p/riscv_test.h

```diff
@@ -64,6 +64,10 @@
 1:
 
 #define INIT_SATP                                                      \
+  csrr t0, misa;    /* Check if S-mode is implemented ... */            \
+  srli t0, t0, 18;                                                      \
+  andi t0, t0, 1;                                                       \
+  beqz t0, 1f;      /* ... Skip this subroutine if it isn't */          \
   la t0, 1f;                                                            \
   csrw mtvec, t0;                                                       \
   csrwi sptbr, 0;                                                       \
```

Comment:
The existing code already handles the no-S-mode case.  The sptbr write will trap, but since mtvec was set to the instruction after sptbr, execution continues normally.